src/fast_sincos.c

```diff
@@ -0,0 +1,475 @@
+        negativeFactor ^= 1;
+    }
+
+    if(clampedAngle >= M_PI_2) {
+        clampedAngle = M_PI - clampedAngle;
+    }
+
+    // Chose the appropriate approximation depending on the
+    // accepted range.
+    fast_sincos_real returnedValue;
+    if(clampedAngle < M_PI_4) {
+        returnedValue = fastSinChebyshev(clampedAngle, degree);
+    } else {
+        returnedValue = fastCosChebyshev(M_PI_2 - clampedAngle, degree);
+    }
+
+    return negativeFactor ? -returnedValue : returnedValue;
+}
+
+/**
+ * @brief Fast cosine computation.
+ * With degrees of 3, 5 and 7, the Chebyshev approximation is used to compute the cosine of the angle.
+ * With degree = 2, the cosine is computed with a lookup sine table and interpolation.
+ * With degree = 1, the cosine is computed with a lookup sine table only.
+ * Invalid degrees will default to 7.
+ * @param angleRadians The angle, in radians
+ * @param degree The degree of the approximation. 
+ * Higher is more accurate, but slower.
+ * @return A cosine approximation of the angle.
+*/
+fast_sincos_real fastCos(const fast_sincos_real angleRadians, const int degree) {
+    
+    // Check for lower degrees first
+    switch (degree) {
+      case 1:
+        return lookupCos(angleRadians);
+      case 2:
+        return lookupCosInterpolate(angleRadians);
+    }
+
+    // Isolate the angle in the first quandrant
+    int negativeFactor = 0;
+    fast_sincos_real clampedAngle = angleRadians;
+    if (clampedAngle < 0) {
+        clampedAngle = -clampedAngle;
+    }
+    
+    clampedAngle = fmod(clampedAngle, PI_TIMES_2);
+    
+    if (clampedAngle >= M_PI) {
+        clampedAngle -= M_PI;
+        negativeFactor ^= 1;
+    }
+
+    if(clampedAngle >= M_PI_2) {
+        clampedAngle = M_PI - clampedAngle;
+        negativeFactor ^= 1;
+    }
+
+    // Chose the appropriate approximation depending on the
+    // accepted range.
+    fast_sincos_real returnedValue;
+    if(clampedAngle < M_PI_4) {
+        returnedValue = fastCosChebyshev(clampedAngle, degree);
+    } else {
+        returnedValue = fastSinChebyshev(M_PI_2 - clampedAngle, degree);
+    }
+
+    return negativeFactor ? -returnedValue : returnedValue;
+}
+
+/**
+ * 0 is 0
+ * 2Pi is 2^25 (arbitrary choice for easier initial calculations wihtout overflow)
+*/
+fast_sincos_real fastFixedSin(const uint32_t fixedAngle) {
+  //const uint32_t limit = 2147483648;//(1 << 31); // 0x7FFF FFFF
+  //fixedAngle &= ~limit; // only get 31 bits
+  //const uint32_t limit = 0x7FFFFFFF;
+  //uint32_t inputAngle = fixedAngle; // & limit;
+
+  // 1<<25 = 33 554 432
+
+  // 22 + 9 = 31
+  const uint16_t RemainderMask = 0xFFFF; // 16 lsb bits
+  const uint16_t indexMask = 0x1FF; // 9 last bits
+
+  //uint16_t remainder = (fixedAngle >> 6) & RemainderMask; // 22 lsb bits
+  //uint16_t index = (((uint16_t*)&fixedAngle)[1] >> 6) & indexMask;  // 9 last bits
+  uint16_t remainder = (((uint16_t*)&fixedAngle)[0]) & RemainderMask; // 16 lsb bits
```

Comment:
No need to cast as a uint16_t* and access the first element, you can simply type cast it as a uint16_t